File: samples/csharp_dotnetcore/82.skills-sso-cloudadapter/SkillBot/Bots/SkillBot.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Threading;
using System.Threading.Tasks;
using Microsoft.Bot.Builder;
using Microsoft.Bot.Builder.Dialogs;

namespace Microsoft.BotBuilderSamples.SkillBot.Bots
{
#pragma warning disable CA1724 // Type names should not match namespaces (by design and we can't change this without breaking binary compat).
    public class SkillBot<T> : ActivityHandler
#pragma warning restore CA1724
        where T : Dialog
    {
        private readonly ConversationState _conversationState;
        private readonly Dialog _mainDialog;

        public SkillBot(ConversationState conversationState, T mainDialog)
        {
            _conversationState = conversationState;
            _mainDialog = mainDialog;
        }

        public override async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default)
        {
            await _mainDialog.RunAsync(turnContext, _conversationState.CreateProperty<DialogState>("DialogState"), cancellationToken);

            // Save any state changes that might have occurred during the turn.
            await _conversationState.SaveChangesAsync(turnContext, false, cancellationToken);
        }
    }
}

File: archive/samples/java_springboot/81.skills-skilldialog/dialog-skill-bot/src/main/java/com/microsoft/bot/sample/dialogskillbot/bots/SkillBot.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MT License.

package com.microsoft.bot.sample.dialogskillbot.bots;

import java.util.concurrent.CompletableFuture;

// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import com.microsoft.bot.builder.ActivityHandler;
import com.microsoft.bot.builder.ConversationState;
import com.microsoft.bot.builder.TurnContext;
import com.microsoft.bot.dialogs.Dialog;

public class SkillBot<T extends Dialog> extends ActivityHandler {

    private ConversationState conversationState;
    private Dialog dialog;

    public SkillBot(ConversationState conversationState, T mainDialog) {
        this.conversationState = conversationState;
        this.dialog = mainDialog;
    }

    @Override
    public CompletableFuture<Void> onTurn(TurnContext turnContext) {
        return Dialog.run(dialog, turnContext, conversationState.createProperty("DialogState"))
        .thenAccept(result -> {
            conversationState.saveChanges(turnContext, false);
        });
    }

}
